torch/testing/__init__.py

```diff
@@ -102,6 +102,15 @@ def get_all_math_dtypes(device):
 
     return dtypes
 
+def get_all_math_dtypes_and_complex(device):
+    dtypes = get_all_math_dtypes(device) + [torch.complex64, torch.complex128]
+
+    # torch.float16 is a math dtype on cuda but not cpu.
+    if device.startswith('cuda'):
```

Comment:
I think this is redundant since `get_all_math_dtypes` already accepts a device and adds this

Reply:
updated!